<a href="https://colab.research.google.com/github/NIKITA0509-I/LOGISTIC-REGRESSION/blob/main/LOGISTIC_REGRESSION.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:

# Data Preprocessing
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [2]:
# Mounting Google Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# importing Dataset from Local drive
from google.colab import files
Uploaded=files.upload()

Saving Social_Network_Ads (1).csv to Social_Network_Ads (1) (1).csv


In [5]:
# Importing the dataset from Google Drive
dataset = pd.read_csv('/content/Social_Network_Ads (1).csv')

In [6]:
print(dataset)

      User ID  Gender  Age  EstimatedSalary  Purchased
0    15624510    Male   19            19000          0
1    15810944    Male   35            20000          0
2    15668575  Female   26            43000          0
3    15603246  Female   27            57000          0
4    15804002    Male   19            76000          0
..        ...     ...  ...              ...        ...
395  15691863  Female   46            41000          1
396  15706071    Male   51            23000          1
397  15654296  Female   50            20000          1
398  15755018    Male   36            33000          0
399  15594041  Female   49            36000          1

[400 rows x 5 columns]


In [7]:
dataset.isnull().sum()

User ID            0
Gender             0
Age                0
EstimatedSalary    0
Purchased          0
dtype: int64

In [8]:
dataset.describe()

User ID         Age  EstimatedSalary   Purchased
count  4.000000e+02  400.000000       400.000000  400.000000
mean   1.569154e+07   37.655000     69742.500000    0.357500
std    7.165832e+04   10.482877     34096.960282    0.479864
min    1.556669e+07   18.000000     15000.000000    0.000000
25%    1.562676e+07   29.750000     43000.000000    0.000000
50%    1.569434e+07   37.000000     70000.000000    0.000000
75%    1.575036e+07   46.000000     88000.000000    1.000000
max    1.581524e+07   60.000000    150000.000000    1.000000

In [9]:
# Extracting dependent and independent variables:
# Extracting independent variable:
X = dataset.iloc[:, [1,2,3]].values
# Extracting dependent variable:
y = dataset.iloc[:, 4].values


In [10]:
print(X)

[['Male' 19 19000]
 ['Male' 35 20000]
 ['Female' 26 43000]
 ...
 ['Female' 50 20000]
 ['Male' 36 33000]
 ['Female' 49 36000]]


In [11]:
# Encoding Categorical data:
# Encoding the Independent Variable
from sklearn.preprocessing import LabelEncoder
labelencoder_X = LabelEncoder()
X[:, 0] = labelencoder_X.fit_transform(X[:, 0])

In [12]:
print(X)

[[1 19 19000]
 [1 35 20000]
 [0 26 43000]
 ...
 [0 50 20000]
 [1 36 33000]
 [0 49 36000]]


In [13]:
print(y)

[0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 0 0 0 1 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0
 0 1 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0
 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 1 0 1 0 1 1 0 0 0 1 0 0 0 1 0 1
 1 1 0 0 1 1 0 1 1 0 1 1 0 1 0 0 0 1 1 0 1 1 0 1 0 1 0 1 0 0 1 1 0 1 0 0 1
 1 0 1 1 0 1 1 0 0 1 0 0 1 1 1 1 1 0 1 1 1 1 0 1 1 0 1 0 1 0 1 1 1 1 0 0 0
 1 1 0 1 1 1 1 1 0 0 0 1 1 0 0 1 0 1 0 1 1 0 1 0 1 1 0 1 1 0 0 0 1 1 0 1 0
 0 1 0 1 0 0 1 1 0 0 1 1 0 1 1 0 0 1 0 1 0 1 1 1 0 1 0 1 1 1 0 1 1 1 1 0 1
 1 1 0 1 0 1 0 0 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 0 1 1 1 0 1]


In [14]:
# Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

In [15]:
print(X_train)

[[0 44 39000]
 [1 32 120000]
 [0 38 50000]
 [0 32 135000]
 [0 52 21000]
 [0 53 104000]
 [1 39 42000]
 [1 38 61000]
 [0 36 50000]
 [0 36 63000]
 [0 35 25000]
 [1 35 50000]
 [1 42 73000]
 [0 47 49000]
 [0 59 29000]
 [1 49 65000]
 [0 45 131000]
 [0 31 89000]
 [0 46 82000]
 [0 47 51000]
 [1 26 15000]
 [1 60 102000]
 [0 38 112000]
 [1 40 107000]
 [0 42 53000]
 [1 35 59000]
 [1 48 41000]
 [0 48 134000]
 [0 38 113000]
 [1 29 148000]
 [0 26 15000]
 [1 60 42000]
 [1 24 19000]
 [1 42 149000]
 [0 46 96000]
 [1 28 59000]
 [1 39 96000]
 [1 28 89000]
 [1 41 72000]
 [1 45 26000]
 [0 33 69000]
 [0 20 82000]
 [1 31 74000]
 [1 42 80000]
 [0 35 72000]
 [0 33 149000]
 [1 40 71000]
 [0 51 146000]
 [1 46 79000]
 [1 35 75000]
 [1 38 51000]
 [0 36 75000]
 [0 37 78000]
 [1 38 61000]
 [0 60 108000]
 [0 20 82000]
 [1 57 74000]
 [1 42 65000]
 [1 26 80000]
 [1 46 117000]
 [1 35 61000]
 [0 21 68000]
 [0 28 44000]
 [1 41 87000]
 [0 37 33000]
 [1 27 90000]
 [1 39 42000]
 [1 28 123000]
 [0 31 118000]
 [1 25 87000]
 [0

In [16]:
print(X_test)

[[1 30 87000]
 [0 38 50000]
 [1 35 75000]
 [0 30 79000]
 [0 35 50000]
 [1 27 20000]
 [0 31 15000]
 [1 36 144000]
 [0 18 68000]
 [1 47 43000]
 [1 30 49000]
 [0 28 55000]
 [1 37 55000]
 [1 39 77000]
 [1 20 86000]
 [0 32 117000]
 [1 37 77000]
 [1 19 85000]
 [0 55 130000]
 [1 35 22000]
 [0 35 47000]
 [0 47 144000]
 [0 41 51000]
 [1 47 105000]
 [0 23 28000]
 [0 49 141000]
 [0 28 87000]
 [1 29 80000]
 [0 37 62000]
 [0 32 86000]
 [1 21 88000]
 [1 37 79000]
 [1 57 60000]
 [1 37 53000]
 [1 24 58000]
 [1 18 52000]
 [1 22 81000]
 [0 34 43000]
 [0 31 34000]
 [0 49 36000]
 [1 27 88000]
 [1 41 52000]
 [0 27 84000]
 [1 35 20000]
 [0 43 112000]
 [0 27 58000]
 [0 37 80000]
 [0 52 90000]
 [1 26 30000]
 [1 49 86000]
 [0 57 122000]
 [0 34 25000]
 [0 35 57000]
 [1 34 115000]
 [0 59 88000]
 [1 45 32000]
 [0 29 83000]
 [0 26 80000]
 [1 49 28000]
 [1 23 20000]
 [1 32 18000]
 [1 60 42000]
 [1 19 76000]
 [1 36 99000]
 [0 19 26000]
 [1 60 83000]
 [0 24 89000]
 [1 27 58000]
 [1 40 47000]
 [0 42 70000]
 [0 32 1500

In [17]:
print(y_test)

[0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 1 0 0 0 0 0 1 1 0 0 0 0
 0 0 1 0 0 0 0 1 0 0 1 0 1 1 0 0 0 1 1 0 0 1 0 0 1 0 1 0 1 0 0 0 0 1 0 0 1
 0 0 0 0 1 1 1 0 0 0 1 1 0 1 1 0 0 1 0 0 0 1 0 1 1 1]


In [18]:
print(y_train)

[0 1 0 1 1 1 0 0 0 0 0 0 1 1 1 0 1 0 0 1 0 1 0 1 0 0 1 1 1 1 0 1 0 1 0 0 1
 0 0 1 0 0 0 0 0 1 1 1 1 0 0 0 1 0 1 0 1 0 0 1 0 0 0 1 0 0 0 1 1 0 0 1 0 1
 1 1 0 0 1 1 0 0 1 1 0 1 0 0 1 1 0 1 1 1 0 0 0 0 0 1 0 0 1 1 1 1 1 0 1 1 0
 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 1 0 0 0 1 0 1 1 0 1 0 0 0 0 1 0 0 0 1 1 0 0
 0 0 1 0 1 0 0 0 1 0 0 0 0 1 1 1 0 0 0 0 0 0 1 1 1 1 1 0 1 0 0 0 0 0 1 0 0
 0 0 0 0 1 1 0 1 0 1 0 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 1 0 1 1 0 0 0 0 0
 0 1 1 0 0 0 0 1 0 0 0 0 1 0 1 0 1 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 1 1 0 0 0
 0 0 1 0 1 1 0 0 0 0 0 1 0 1 0 0 1 0 0 1 0 1 0 0 0 0 0 0 1 1 1 1 0 0 0 0 1
 0 0 0 0]


In [19]:
# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [20]:
print(X_train)

[[-0.98019606  0.58164944 -0.88670699]
 [ 1.02020406 -0.60673761  1.46173768]
 [-0.98019606 -0.01254409 -0.5677824 ]
 [-0.98019606 -0.60673761  1.89663484]
 [-0.98019606  1.37390747 -1.40858358]
 [-0.98019606  1.47293972  0.99784738]
 [ 1.02020406  0.08648817 -0.79972756]
 [ 1.02020406 -0.01254409 -0.24885782]
 [-0.98019606 -0.21060859 -0.5677824 ]
 [-0.98019606 -0.21060859 -0.19087153]
 [-0.98019606 -0.30964085 -1.29261101]
 [ 1.02020406 -0.30964085 -0.5677824 ]
 [ 1.02020406  0.38358493  0.09905991]
 [-0.98019606  0.8787462  -0.59677555]
 [-0.98019606  2.06713324 -1.17663843]
 [ 1.02020406  1.07681071 -0.13288524]
 [-0.98019606  0.68068169  1.78066227]
 [-0.98019606 -0.70576986  0.56295021]
 [-0.98019606  0.77971394  0.35999821]
 [-0.98019606  0.8787462  -0.53878926]
 [ 1.02020406 -1.20093113 -1.58254245]
 [ 1.02020406  2.1661655   0.93986109]
 [-0.98019606 -0.01254409  1.22979253]
 [ 1.02020406  0.18552042  1.08482681]
 [-0.98019606  0.38358493 -0.48080297]
 [ 1.02020406 -0.30964085

In [21]:
from sklearn.linear_model  import LogisticRegression
classifier = LogisticRegression(random_state=0)
classifier.fit(X_train, y_train)



LogisticRegression(random_state=0)

In [22]:
# Predicting the Test set results
y_pred = classifier.predict(X_test)
y_pred

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1])

In [23]:
# Making the Confusion Matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print('Confusion matrix: ')
print(cm)

Confusion matrix: 
[[65  3]
 [ 7 25]]


In [24]:
# calculate Accuracy
#               Y_pred
#             0       1
#          0[64     4(FP)]  Type 1 Error
# y-test   1[4(FN)     28]  Type 2 Error

from sklearn.metrics import accuracy_score
print('Accuracy: %.2f' % (accuracy_score(y_test, y_pred)*100))

Accuracy: 90.00


In [25]:
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score
# calculate precision
# Precision = TruePositives / (TruePositives + FalsePositives)
#               Y_pred
#             0       1
#          0[64     4(FP)]  Type 1 Error
# y-test   1[4(FN)     28]  Type 2 Error
precision = precision_score(y_test, y_pred, average='binary')
print('Precision: %.3f' % (precision*100))

Precision: 89.286


In [26]:
# calculate recall
# Recall = TruePositives / (TruePositives + FalseNegatives)
#               Y_pred
#             0       1
#          0[64     4(FP)]  Type 1 Error
# y-test   1[4(FN)     28]  Type 2 Error
recall = recall_score(y_test, y_pred, average='binary')
print('Recall: %.3f' % (recall*100))

Recall: 78.125


In [27]:
# F-Measure = (2 * Precision * Recall) / (Precision + Recall)
# calculate score
score = f1_score(y_test, y_pred, average='binary')
print('F-Measure: %.3f' % (score*100))

F-Measure: 83.333


In [28]:
age = 42 #@param {type:"number"}
gender =  1#@param {type:"number"}
Estimated_Salary = 1500 #@param {type:"number"}
deepak =  2#@param {type:"number"}
output=classifier.predict(sc.transform([[ gender,age ,Estimated_Salary ]]))
print("Purchased =", output)
if output==[1]:
  print( 'Custome will purchase Product')
else:
  print('Customer will not purchase computer')


Purchased = [0]
Customer will not purchase computer


In [29]:
import pickle 
  
# Save the trained model as a pickle string. 
saved_model = pickle.dumps(classifier) 
  
# Load the pickled model 
model = pickle.loads(saved_model) 
  
# Use the loaded pickled model to make predictions 
model.predict(X_test) 

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1])

In [30]:
# save the model to disk
import joblib
filename = '/content/Social_Network_Ads (1).sav'
joblib.dump(classifier, filename)
 #-----------------------------------------------------------
# some time later...
 
# load the model from disk
randomforest_model = joblib.load(filename)
result = randomforest_model.score(X_test, y_test)
print(result)

0.9


In [35]:
import pickle 
print("[INFO] Saving model...")
# Save the trained model as a pickle string. 
saved_model=pickle.dump(classifier,open('/content/Social_Network_Ads (1).pkl', 'wb')) 
# Saving model to disk

[INFO] Saving model...


In [36]:
# Load the pickled model 
model = pickle.load(open('/content/Social_Network_Ads (1).pkl', 'rb'))  
# Use the loaded pickled model to make predictions 
model.predict(X_test)

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1,
       0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1])

In [37]:
# Mounting Google Drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [38]:
!pip install streamlit

     |████████████████████████████████| 10.1 MB 29.9 MB/s 
     |████████████████████████████████| 164 kB 56.6 MB/s 
     |████████████████████████████████| 181 kB 18.9 MB/s 
     |████████████████████████████████| 4.3 MB 49.5 MB/s 
     |████████████████████████████████| 111 kB 71.8 MB/s 
     |████████████████████████████████| 76 kB 6.9 MB/s 
     |████████████████████████████████| 63 kB 2.3 MB/s 
     |████████████████████████████████| 131 kB 77.9 MB/s 
     |████████████████████████████████| 428 kB 62.5 MB/s 
     |████████████████████████████████| 130 kB 55.2 MB/s 
     |████████████████████████████████| 793 kB 65.5 MB/s 
     |████████████████████████████████| 381 kB 67.4 MB/s 
  Created wheel for blinker: filename=blinker-1.4-py3-none-any.whl size=13478 sha256=2e4d979f17fd67a28a63f414e2cb8af50cc1b2645077f973d79bebaa10e81c06
  Stored in directory: /root/.cache/pip/wheels/22/f5/18/df711b66eb25b21325c132757d4314db9ac5e8dabeaf196eab
Successfully built blinker
  Attempting uninstall:

In [39]:
# Mounting Google Drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [40]:
!pip install pyngrok

     |████████████████████████████████| 745 kB 7.4 MB/s 
  Created wheel for pyngrok: filename=pyngrok-5.1.0-py3-none-any.whl size=19007 sha256=2bc276a969641d7d2ab8960a2ecf1806f8931f9ce8d968fa8922315fe1680e36
  Stored in directory: /root/.cache/pip/wheels/bf/e6/af/ccf6598ecefecd44104069371795cb9b3afbcd16987f6ccfb3
Successfully built pyngrok


In [41]:
!ngrok authtoken 24ustNzCIlDOkO3r5iUx6tCUolz_37zoKr3tq8hZAppzJxd95


Authtoken saved to configuration file: /root/.ngrok2/ngrok.yml


In [47]:
%%writefile app.py
import streamlit as st 
#from PIL import Image
import pickle
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
st.set_option('deprecation.showfileUploaderEncoding', False)
# Load the pickled model
model = pickle.load(open('/content/Social_Network_Ads (1).pkl', 'rb')) 
# Feature Scaling
dataset = pd.read_csv('/content/Social_Network_Ads (1).csv')
# Extracting independent variable:
X = dataset.iloc[:, [1,2,3]].values
# Encoding the Independent Variable
from sklearn.preprocessing import LabelEncoder
labelencoder_X = LabelEncoder()
X[:, 0] = labelencoder_X.fit_transform(X[:, 0])
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X = sc.fit_transform(X)
def predict_note_authentication(UserID, Gender,Age,EstimatedSalary):
  output= model.predict(sc.transform([[Gender,Age,EstimatedSalary]]))
  print("Purchased", output)
  if output==[1]:
    prediction="Item will be purchased"
  else:
    prediction="Item will not be purchased"
  print(prediction)
  return prediction
def main():
    
    html_temp = """
   <div class="" style="background-color:Brown;" >
   <div class="clearfix">           
   <div class="col-md-12">
   <center><p style="font-size:40px;color:black;margin-top:10px;">Poornima Institute of Engineering & Technology</p></center> 
   <center><p style="font-size:30px;color:black;margin-top:10px;">Department of Computer Engineering</p></center> 
   <center><p style="font-size:25px;color:black;margin-top:10px;"Machine Learning Lab Experiment</p></center> 
   </div>
   </div>
   </div>
   """
    st.markdown(html_temp,unsafe_allow_html=True)
    st.header("Item Purchase Prediction using Logistic Classification")
    
    UserID = st.text_input("UserID","")
    
    #Gender1 = st.select_slider('Select a Gender Male:1 Female:0',options=['1', '0'])
    #Gender1 = st.number_input('Insert Gender Male:1 Female:0')
    Gender1 = st.text_input('Insert Gender Male:1 Female:0')
    Age = st.number_input('Insert a Age',18,60)
   
    EstimatedSalary = st.number_input("Insert Estimated Salary",15000,150000)
    resul=""
    if st.button("Predict"):
      result=predict_note_authentication(UserID, Gender1,Age,EstimatedSalary)
      st.success('Model has predicted {}'.format(result))
      
    if st.button("About"):
      st.subheader("Developed by NIKITA KHANDAL")
      st.subheader("STUDENT , Department of Computer Engineering")

if __name__=='__main__':
  main()

Overwriting app.py


In [48]:
!nohup streamlit run  app.py &

nohup: appending output to 'nohup.out'


In [49]:
from pyngrok import ngrok
url=ngrok.connect(port='8050')
url

<NgrokTunnel: "http://bc1b-34-143-223-145.ngrok.io" -> "http://localhost:80">

In [ ]:
!streamlit run --server.port 80 app.py

2022-04-23 06:16:44.824 INFO    numexpr.utils: NumExpr defaulting to 2 threads.

  You can now view your Streamlit app in your browser.

  Network URL: http://172.28.0.2:80
  External URL: http://34.143.223.145:80



In [ ]:
ngrok.kill()

Model Deployment

Heroku has a pretty good free tier service, and lets you host up to 5 unverified apps at once (unverified means you haven’t given them a credit card), each with a custom domain.

First you will need a requirements file that tells Heroku which Python libraries are required for your app, and which versions. The easiest way to generate this file is with the pipreqs library. You can install pipreqs with either pip or anaconda.

Flask==1.1.1

gunicorn==19.9.0

itsdangerous==1.1.0

Jinja2==2.10.1

MarkupSafe==1.1.1

Werkzeug==0.15.5

numpy>=1.9.2

scipy>=0.15.1

scikit-learn==0.24.1

matplotlib>=1.4.3

pandas>=0.19

streamlit

Next you will need a bash script to configure Streamlit for use on Heroku. The file should be named “setup.sh” and should have the exact code below. You can edit the file in any text editor. Again, this needs to be in your project’s main directory.

mkdir -p ~/.streamlit

echo "[server]

headless = true

port = $PORT

enableCORS = false

" > ~/.streamlit/config.toml

Lastly, you need an odd file called a Procfile, which specifies the commands that are executed by the app on startup. It should be spelled exactly as “Procfile”, without any file extension, and should be in your project’s main directory. You can edit it with any text editor. Replace the last part of the code below with your Streamlit python script.

web: sh setup.sh && streamlit run your-script.py

Now all of the necessary files should be in your local repo: your python script (for Streamlit)

Procfile

setup.sh

requirements.txt

Any files that are read into the python script need to be in the directory as well.

